### PR TITLE
Remove conflicting sqlite tables that are "reserved" (shadow fts4 tables)

### PR DESCRIPTION
**Pull request by MadLittleMods**
Remove conflicting sqlite tables that throw `sqlite3.OperationalError: object name reserved for internal use: event_search_content` when running the twisted unit tests.

Fix https://github.com/matrix-org/synapse/issues/8996

If I remove these tables, the tests seem to run well 🤷

Before:

```sh
$ python -m twisted.trial tests
...
sqlite3.OperationalError: object name reserved for internal use: event_search_content

Ran 1412 tests in 57.728s

FAILED (skips=20, errors=1069, successes=323)
```

After:

```sh
$ python -m twisted.trial tests
...
Ran 1412 tests in 704.890s

PASSED (skips=20, successes=1392)
```

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: synapse/storage/databases/main/schema/full_schemas/54/full.sql.sqlite [hunk 1]
```diff
@@ -67,11 +67,6 @@ CREATE TABLE IF NOT EXISTS "user_threepids" ( user_id TEXT NOT NULL, medium TEXT
 CREATE INDEX user_threepids_user_id ON user_threepids(user_id);
 CREATE VIRTUAL TABLE event_search USING fts4 ( event_id, room_id, sender, key, value )
 /* event_search(event_id,room_id,sender,"key",value) */;
-CREATE TABLE IF NOT EXISTS 'event_search_content'(docid INTEGER PRIMARY KEY, 'c0event_id', 'c1room_id', 'c2sender', 'c3key', 'c4value');
```
Comment (MadLittleMods) [inline synapse/storage/databases/main/schema/full_schemas/54/full.sql.sqlite:70]: Are these tables used anymore? I can't find any references to them and these are the tables causing `sqlite3.OperationalError: object name reserved for internal use: event_search_content` errors.

**Comment by richvdh** [comment]
see #8996 for some background on this.

also: If we're removing things from the schema files, we should also update the script that is used to dump a new full schema so we don't have the same problem next time.

File: synapse/storage/databases/main/schema/full_schemas/54/full.sql.sqlite (hunk 1 above)
Comment (MadLittleMods) [reply synapse/storage/databases/main/schema/full_schemas/54/full.sql.sqlite:70]: This was clarified by @richvdh in https://github.com/matrix-org/synapse/issues/8996#issuecomment-754271226. These are shadow tables from the fts4 virtual table above.

File: scripts-dev/make_full_schema.sh [hunk 2]
```diff
@@ -162,12 +162,23 @@ else
...
+DROP TABLE event_search_content;
+DROP TABLE event_search_segments;
+DROP TABLE event_search_segdir;
+DROP TABLE event_search_docsize;
+DROP TABLE event_search_stat;
+DROP TABLE user_directory_search_content;
+DROP TABLE user_directory_search_segments;
+DROP TABLE user_directory_search_segdir;
+DROP TABLE user_directory_search_docsize;
+DROP TABLE user_directory_search_stat;
```
Comment (MadLittleMods) [inline scripts-dev/make_full_schema.sh:181]: For reference, there is documentation about this script at [`synapse/storage/databases/main/schema/full_schemas/README.md`](https://github.com/matrix-org/synapse/blob/d2c616a41381c9e2d43b08d5f225b52042d94d23/synapse/storage/databases/main/schema/full_schemas/README.md)

File: scripts-dev/make_full_schema.sh [hunk 3]
```diff
@@ -162,12 +162,23 @@ else
...
```
Comment (MadLittleMods) [inline scripts-dev/make_full_schema.sh:181]: I haven't tested these changes as the script seems to fail for me before it gets there. Also is there a way to run without a postgres username? (I had to comment that part of the shell script out)


```
$ ./scripts-dev/make_full_schema.sh -o ./
Creating port tables...
Setting up sequence generators...
scripts/synapse_port_db:630: RuntimeWarning: coroutine 'DatabasePool.runInteraction' was never awaited
  await self._setup_user_id_seq()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
2021-01-04 17:48:57,303 - synapse_port_db - 704 - ERROR -
Traceback (most recent call last):
  File "scripts/synapse_port_db", line 631, in run
    await self._setup_events_stream_seqs()
  File "scripts/synapse_port_db", line 910, in _setup_events_stream_seqs
    return await self.postgres_store.db_pool.runInteraction(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 632, in runInteraction
    result = await self.runWithConnection(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 712, in runWithConnection
    return await make_deferred_yieldable(
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 306, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/compat.py", line 464, in reraise
    raise exception.with_traceback(traceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 297, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 707, in inner_func
    return func(db_conn, *args, **kwargs)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 510, in new_transaction
    r = func(cursor, *args, **kwargs)
  File "scripts/synapse_port_db", line 907, in _setup_events_stream_seqs_set_pos
    (curr_backward_id + 1,),
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
Traceback (most recent call last):
  File "scripts/synapse_port_db", line 631, in run
    await self._setup_events_stream_seqs()
  File "scripts/synapse_port_db", line 910, in _setup_events_stream_seqs
    return await self.postgres_store.db_pool.runInteraction(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 632, in runInteraction
    result = await self.runWithConnection(
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 712, in runWithConnection
    return await make_deferred_yieldable(
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 306, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/python/compat.py", line 464, in reraise
    raise exception.with_traceback(traceback)
  File "/Users/eric/Documents/github/synapse/env/lib/python3.8/site-packages/twisted/enterprise/adbapi.py", line 297, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 707, in inner_func
    return func(db_conn, *args, **kwargs)
  File "/Users/eric/Documents/github/synapse/synapse/storage/database.py", line 510, in new_transaction
    r = func(cursor, *args, **kwargs)
  File "scripts/synapse_port_db", line 907, in _setup_events_stream_seqs_set_pos
    (curr_backward_id + 1,),
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```

**Review by richvdh — COMMENTED** [review]
this is fine, but I'm not thrilled about it. It's a bit magical, and will break if we add more virtual tables in the future. I'd really like to get to the bottom of the problem.

File: changelog.d/9003.misc [hunk 4]
```diff
@@ -0,0 +1 @@
+Remove fts4 shadow tables from schema dumps.
```
Comment (richvdh) [inline changelog.d/9003.misc:1]: Could you update this to describe what this means for the user?

"Fix 'object name reserved for internal use' with recent versions of SQLite." or so?

**Review by richvdh — APPROVED** [review]
looks good, other than the changelog.

File: changelog.d/9003.misc (hunk 4 above)
Comment (MadLittleMods) [reply changelog.d/9003.misc]: Added 🚀

**Review by richvdh — APPROVED** [review]

**Comment by MadLittleMods** [comment]
Thanks @richvdh! Glad you were able to find the [root cause](https://github.com/matrix-org/synapse/issues/8996#issuecomment-776933455) on why this was necessary :D

**Pull request merged**